diff -r ce13b7c6659c wokkel/disco.py
--- a/wokkel/disco.py	Sun Dec 26 23:15:41 2010 +0100
+++ b/wokkel/disco.py	Tue Feb 08 01:16:34 2011 +0100
@@ -445,8 +445,14 @@
         @param iq: The request iq element.
         @type iq: L{Element<twisted.words.xish.domish.Element>}
         """
-        requestor = jid.internJID(iq["from"])
-        target = jid.internJID(iq["to"])
+        try:
+            requestor = jid.internJID(iq["from"])
+        except KeyError:
+            requestor = None
+        try:
+            target = jid.internJID(iq["to"])
+        except KeyError:
+            target = None
         nodeIdentifier = iq.query.getAttribute("node", '')
 
         def toResponse(info):
